client/landing/stepper/hooks/use-update-flow-progress.ts

```diff
@@ -0,0 +1,146 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { Onboard } from '@automattic/data-stores';
+import { useIsEnglishLocale } from '@automattic/i18n-utils';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { ONBOARD_STORE } from '../stores';
+import type { StepPath } from '../declarative-flow/internals/steps-repository';
+
+const SiteIntent = Onboard.SiteIntent;
+const MAX_STEPS = 10;
+
+export function useUpdateFlowProgress( currentStep: StepPath ) {
```

Comment:
My two cents: This hook should only handle the logic of resolving the flow progress, but shouldn't update the Redux store. In other words, instead of directly setting `setStepProgress( { progress: flowProgress, count: MAX_STEPS } );`, we can return the object `{ progress: flowProgress, count: MAX_STEPS }`.

Reply:
Will do